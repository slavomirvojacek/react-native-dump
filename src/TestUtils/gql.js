const assertQueryOrMutation = (QoM) => {
  test('matches Query or Mutation', () => {
    expect(QoM).toMatchSnapshot()
  })
}

const assertGql = (Component) => {
  const { gql } = Component
  expect(gql instanceof Object).toEqual(true)
  Object
    .keys(gql)
    .forEach(k => assertQueryOrMutation(gql[k]))
}

export { assertGql }
