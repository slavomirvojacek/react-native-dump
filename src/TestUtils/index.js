export { assertGql } from './gql'
export { mockSetState } from './mocking'
export { assertSnapshots } from './snapshot'
