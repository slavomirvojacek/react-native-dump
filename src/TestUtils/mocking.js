// Sometimes, we might want to test internal state changes without having to render.
// This overrides the setState method to only update the state, without interacting with
// React's internals such as the scheduler.
const mockSetState = (Component) => {
  return class extends Component {
    setState (f) {
      if (typeof f === 'function') {
        this.state = {...this.state, ...f(this.state)}
      } else {
        this.state = {...this.state, ...f}
      }
    }
  }
}

export { mockSetState }
