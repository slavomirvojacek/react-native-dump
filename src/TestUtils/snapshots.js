import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

const assertSnapshots = (Component, props, desc, store = null) => {
  let getComponent = p => null

  if (store !== null) {
    getComponent = p => (
      <Provider store={store}>
        <Component {...p}/>
      </Provider>
    )
  } else {
    getComponent = p => <Component {...p}/>
  }

  test(desc, () => {
    const tree = renderer
      .create(getComponent(props))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
}

export { assertSnapshots }
