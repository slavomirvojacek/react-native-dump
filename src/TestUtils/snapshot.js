import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

import { assertGql } from './gql'

import { isNull } from '../Utils/type'

const assertSnapshot = (Component, props, desc, store = null) => {
  let getComponent = p => null

  // In theory, if things are done correctly, this (wrapping
  // a component in Provider with a store) should never be needed.
  if (!isNull(store)) {
    getComponent = p => (
      <Provider store={store}>
        <Component {...p}/>
      </Provider>
    )
  } else {
    getComponent = p => <Component {...p}/>
  }

  test(desc, () => {
    const tree = renderer
      .create(getComponent(props))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
}

const assertSnapshots = (Component, configs, store = null) => {
  configs.forEach(config => assertSnapshot(Component, config.props, config.desc, store))
  if (!isNull(Component.gql)) assertGql(Component)
}

export { assertSnapshots }
