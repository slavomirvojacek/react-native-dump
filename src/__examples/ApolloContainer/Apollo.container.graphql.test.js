import { ApolloContainer } from './Apollo.container'

import { assertGql } from '../../TestUtils/gql'

describe('ApolloContainer GraphQL Config', () => {
  assertGql(ApolloContainer)
})
