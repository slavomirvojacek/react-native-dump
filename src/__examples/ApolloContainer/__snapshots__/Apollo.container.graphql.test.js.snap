// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApolloContainer GraphQL Config matches Query or Mutation 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": undefined,
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "owner",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "repoOwner",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "name",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "repoName",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "repository",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [
                    Object {
                      "kind": "Argument",
                      "name": Object {
                        "kind": "Name",
                        "value": "last",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "limit",
                        },
                      },
                    },
                  ],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "issues",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": null,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "nodes",
                        },
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "id",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "title",
                              },
                              "selectionSet": null,
                            },
                            Object {
                              "alias": null,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "url",
                              },
                              "selectionSet": null,
                            },
                          ],
                        },
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "repoOwner",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "repoName",
            },
          },
        },
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NamedType",
            "name": Object {
              "kind": "Name",
              "value": "Int",
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "limit",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 216,
    "start": 0,
  },
}
`;

exports[`ApolloContainer GraphQL Config matches Query or Mutation 2`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": undefined,
      "operation": "mutation",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": null,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "arg",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "arg",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "something",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": null,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": null,
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": null,
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "arg",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 93,
    "start": 0,
  },
}
`;
